app/controllers/testOnly/ReportOneLoginProblemController.scala

```diff
@@ -0,0 +1,164 @@
+          s"problem_report.email.error.valid",
+          email => emailValidator.validate(email) || email.isEmpty
+        )
+        .verifying("problem_report.email.error.length", email => email.length <= 255),
+      "report-action" -> text
+        .verifying(
+          s"problem_report.action.error.required",
+          action => action.nonEmpty
+        )
+        .verifying("problem_report.action.error.length", action => action.length <= 1000),
+      "report-error" -> text
+        .verifying(
+          s"problem_report.error.error.required",
+          error => error.nonEmpty
+        )
+        .verifying("problem_report.error.error.length", error => error.length <= 1000),
+      "isJavascript" -> boolean,
+      "service"       -> optional(text),
+      "referrer" -> optional(text),
+      "csrfToken" -> text,
+      "userAction" -> optional(text)
+    )(ReportProblemForm.apply)(o => Some(Tuple.fromProductTyped(o)))
+  )
+
+  def emptyForm(csrfToken: String, service: Option[String], referrer: Option[String]): Form[ReportProblemForm] =
+    ReportOneLoginProblemFormBind.form.fill(
+      ReportProblemForm(
+        reportName = "",
+        reportEmail = "",
+        reportAction = "",
+        reportError = "",
+        isJavascript = false,
+        service = Some("one-login-complaint"),
+        referrer = referrer,
+        csrfToken = csrfToken,
+        userAction = None
+      )
+    )
+}
+
+@Singleton
+class ReportOneLoginProblemController @Inject() (
+  val ticketQueueConnector: DeskproTicketQueueConnector,
+  mcc: MessagesControllerComponents,
+  reportOneLoginProblemPage: ReportOneLoginProblemPage,
+  oneLoginConfirmationPage: ReportOneLoginConfirmationPage,
+  errorPage: InternalErrorPage,
+  headerRetriever: RefererHeaderRetriever
+)(using AppConfig, ExecutionContext)
+  extends FrontendController(mcc)
+    with DeskproSubmission
+    with I18nSupport {
+
+  def index(referrerUrl: Option[ReferrerUrl]): Action[AnyContent] = Action { request =>
+    given Request[AnyContent] = request
+
+    val csrfToken = play.filters.csrf.CSRF.getToken(request).map(_.value).getOrElse("")
+    val referrer = referrerUrl orElse headerRetriever.refererFromHeaders()
+
+    Ok(page(ReportOneLoginProblemFormBind.emptyForm(csrfToken, Some("one-login-complaint"), referrer), referrerUrl))
+  }
+
+  def submit(referrerUrl: Option[ReferrerUrl]): Action[AnyContent] =
+    Action.async { request =>
+      given MessagesRequest[AnyContent] = request
+
+      doSubmit(referrerUrl)
+    }
+
+  private def doSubmit(referrerUrl: Option[String])(using request: MessagesRequest[AnyContent]) =
+    ReportOneLoginProblemFormBind.form
+      .bindFromRequest()
+      .fold(
+        formWithError =>
+          Future.successful(
+            BadRequest(page(formWithError, referrerUrl.orElse(fromForm("referrer", formWithError))))
+          ),
+        report => {
+          val problemReport = report.copy(service = Some("one-login-complaint"))
```

Comment:
On Submit before doing backend call I am copying the form to ensure the service is set to what it should be